src/rest-server/src/controllers/v2/job.js

```diff
@@ -17,29 +17,29 @@
 
 
 // module dependencies
-const util = require('util');
 const yaml = require('js-yaml');
 const status = require('statuses');
 const asyncHandler = require('../../middlewares/v2/asyncHandler');
-const v1Job = require('../../models/job');
-const {put, getJobConfig} = require('../../models/v2/job');
+const job = require('../../models/v2/job');
 const createError = require('../../util/error');
 
 
 const update = asyncHandler(async (req, res) => {
   const jobName = res.locals.protocol.name;
   const userName = req.user.username;
   const frameworkName = `${userName}~${jobName}`;
+  // check duplicate job
   try {
-    const getJobError = await util.promisify(v1Job.prototype.getJob)(jobName, userName);
-    if (getJobError.code !== 'NoJobError') {
+    const data = await job.get(frameworkName);
+    if (data != null) {
       throw createError('Conflict', 'ConflictJobError', `Job ${frameworkName} already exists.`);
     }
   } catch (error) {
-    throw createError('Conflict', 'ConflictJobError', `Job ${frameworkName} already exists.`);
+    if (error.code === 'UnknownError') {
```

Comment:
Only throw error when error code is UnknownError? It's an expect behavior?

Reply:
yes

Reply:
updated